core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -747,6 +747,10 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         listing.synchronized {
           listing.delete(classOf[LogInfo], rootPath.toString)
         }
+      case _: FileNotFoundException
+          if reader.rootPath.getName.endsWith(EventLogFileWriter.IN_PROGRESS) =>
+        logInfo(s"In-progress file does not exist: ${reader.rootPath}. The application may be " +
+          "completed during processing.")
```

Comment:
If Spark checks the final location, we can remove this warning.
> ... The application may be completed during processing.

And, if Spark cannot find the log file at the final locations, it could be removed by the users instead of `completed`.

Reply:
Just an idea. Please let me know the difficulty.

Reply:
Yes, simply (1).

Reply:
+1